### ISSUE TITLE
Need to write output files in the outputs directory

### ISSUE THREAD

**Issue by shahab122** [CIROH-UA/NGIAB-CloudInfra#135]
I am running NGIAB on blackwarrior.ua.edu using the latest data obtained from this command: wget --no-parent https://ciroh-ua-ngen-data.s3.us-east-2.amazonaws.com/AWI-004/AWI_09_004.tar.gz. Everything seems to work fine except that the output files from the t-route simulations are written somewhere else other than the 'outputs' directory. 

## Current behavior
Output files are created in root of input, not outputs directory.

## Expected behavior
Output files should be in the outputs directory.

## Steps to replicate behavior (include URLs)

1.

## Screenshots
![image](https://github.com/CIROH-UA/NGIAB-CloudInfra/assets/28275758/47a1819f-7c21-454b-b025-2132a5273afd)

**Comment by arpita0911patel** [comment]
We noticed that output files are created inside the container at /ngen/ngen/data path instead of outputs folder. Causing the 0 output files in outputs folder.

**Comment by arpita0911patel** [comment]
Zach Wills
  [12:14 PM](https://cirohworkspace.slack.com/archives/D04CBML7AKA/p1711386840235999)
Probably need to set an output root directory like described here: https://github.com/NOAA-OWP/ngen/pull/531

**Comment by arpita0911patel** [comment]
It worked on my end after adding output_root path in realization file as below:
 },
 "time": {
  "start_time": "2024-02-23 01:00:00",
  "end_time": "2024-02-24 00:00:00",
  "output_interval": 3600
 },
 "catchments": {}**,
 "output_root": "/ngen/ngen/data/outputs"**
}

**Comment by taddyb** [comment]
There are still problems. Here is the realization I used:

```json
{
  "global": {
    "formulations": [
      {
        "name": "bmi_multi",
        "params": {
          "name": "bmi_multi",
          "model_type_name": "bmi_multi",
          "main_output_variable": "Q_OUT",
          "init_config": "",
          "allow_exceed_end_time": false,
          "fixed_time_step": false,
          "uses_forcing_file": false,
          "modules": [
            {
              "name": "bmi_c++",
              "params": {
                "name": "bmi_c++",
                "model_type_name": "SLOTH",
                "main_output_variable": "z",
                "init_config": "/dev/null",
                "allow_exceed_end_time": true,
                "fixed_time_step": false,
                "uses_forcing_file": false,
                "model_params": {
                  "sloth_ice_fraction_schaake(1,double,m,node)": "0.0",
                  "sloth_ice_fraction_xinanjiang(1,double,1,node)": "0.0",
                  "sloth_smp(1,double,1,node)": "0.0",
                  "EVAPOTRANS": "0.0"
                },
                "library_file": "/dmod/shared_libs/libslothmodel.so",
                "registration_function": "none"
              }
            },
            {
              "name": "bmi_c",
              "params": {
                "name": "bmi_c",
                "model_type_name": "CFE",
                "main_output_variable": "Q_OUT",
                "init_config": "./config/config.ini",
                "allow_exceed_end_time": true,
                "fixed_time_step": false,
                "uses_forcing_file": false,
                "variables_names_map": {
                  "atmosphere_water__liquid_equivalent_precipitation_rate": "precip_rate",
                  "water_potential_evaporation_flux": "EVAPOTRANS",
                  "ice_fraction_schaake": "sloth_ice_fraction_schaake",
                  "ice_fraction_xinanjiang": "sloth_ice_fraction_xinanjiang",
                  "soil_moisture_profile": "sloth_smp"
                },
                "model_params": {
                  "b": 8.660529385231255,
                  "satdk": 0.00011760880965802808,
                  "maxsmc": 0.543673362985325,
                  "refkdt": 3.6613440504586134,
                  "slope": 0.8154788969461678,
                  "max_gw_storage": 0.04021994414923359,
                  "expon": 7.308820146231674,
                  "Cgw": 0.0004609207383395736,
                  "Klf": 0.1681695665829872,
                  "Kn": 0.4017865685354076
                },
                "library_file": "/dmod/shared_libs/libcfebmi.so.1.0.0",
                "registration_function": "register_bmi_cfe"
              }
            }
          ]
        }
      }
    ],
    "forcing": {
      "file_pattern": ".*{{id}}.*.csv",
      "path": "./forcings",
      "provider": "CsvPerFeature"
    }
  },
  "time": {
    "start_time": "2024-02-23 01:00:00",
    "end_time": "2024-02-24 00:00:00",
    "output_interval": 3600
  },
  "catchments": {},
  "output_root": "/ngen/ngen/data/outputs"  
}
```

and the latest error message

```sh
(base) taddbindas in ~/projects/NGIAB-CloudInfra on main ● ● λ ./guide.sh 

=========================================================
 Welcome to CIROH-UA:NextGen National Water Model App! 
=========================================================

Looking for input data (a directory containing the following directories: forcings, config and outputs):

forcings is the hydrofabric input data for your model(s).
config folder has all the configuration related files for the model.
outputs is where the output files are copied to when the model finish the run


Last used data directory path: /Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data
Do you want to use the same path? (Y/n): Y
The Directory you've given is:
/Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data

forcings exists.    11000 forcings found.
config exists.        4 config found.
outputs exists.        0 outputs found.
Files found:        0
Outputs is ready for run. No matching files found.
Files found:        0
ngen/data is ready for run. No matching files found.

Looking in the provided directory gives us:
Found these datastream files:
/Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data/config/datastream.gpkg
Found these datastream files:
/Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data/config/datastream.gpkg
Found these realization files:
/Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data/config/realization.json

Detected ISA = Darwin Computer-air.local 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:34 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8103 arm64
Docker version 24.0.7, build afdd53b4e3
Docker found
Darwin Computer-air.local 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:34 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8103 arm64
Select an option (type a number): 
1) Run NextGen using existing local docker image
2) Run NextGen after updating to latest docker image
3) Exit
#? 2
pulling container and running the model
latest: Pulling from awiciroh/ciroh-ngen-image
Digest: sha256:8f4f7786b7ff24f30ca4ff655ee3d2537996e4fb29a818997afb40e226eecb45
Status: Image is up to date for awiciroh/ciroh-ngen-image:latest
docker.io/awiciroh/ciroh-ngen-image:latest

Running NextGen docker container...
Mounting local host directory /Users/taddbindas/projects/NGIAB-CloudInfra/NextGen/ngen-data/my_data to /ngen/ngen/data within the container.
Working directory is:
/ngen/ngen/data


Found these Catchment files:
./config/datastream.gpkg


Found these Nexus files:
./config/datastream.gpkg


Found these Realization files:
./config/realization.json


Entering Interactive Mode
./HelloNGEN.sh: line 75: continue: only meaningful in a `for', `while', or `until' loop
Select an option (type a number): 
1) Run NextGen model framework in serial mode
2) Run NextGen model framework in parallel mode
3) Run Bash shell
4) Exit
#? 2


Selected files:
Catchment: ./config/datastream.gpkg
Nexus: ./config/datastream.gpkg
Realization: ./config/realization.json

Reading 10999 features from layer divides using ID column `divide_id`
Partitioning 10999 catchments into 2 partitions.
Reading 6232 features from layer nexus using ID column `id`
Validating catchments...

Number of catchments is: 10999
Catchment validation completed
Found 20 remotes in partition 0
Found 33 remotes in partition 1
Found 53 total remotes (average of approximately 26 remotes per partition)
Your NGEN run command is mpirun -n 2 /dmod/bin/ngen-parallel ./config/datastream.gpkg all ./config/datastream.gpkg all ./config/realization.json /ngen/ngen/data/partitions_2.json
If your model didn't run, or encountered an error, try checking the Forcings paths in the Realizations file you selected.
Do you want to redirect command output to /dev/null? (y/N, default: n):
n
```

error:
```sh
Finished function parsing CFE config
At declaration of smc_profile size, soil_reservoir.n_soil_layers = 0
terminate called after throwing an instance of 'std::runtime_error'
  what():  Errno 24 (Too many open files) opening ./forcings/cat-1487413.csv
Config file details - Line Count: 27 | Max Line Length 46
Config Value - Param: 'forcing_file' | Value: 'BMI' | Units: '(null)'
Config Value - Param: 'surface_partitioning_scheme' | Value: 'Schaake' | Units: '(null)'
Config Value - Param: 'soil_params.depth' | Value: '2.0' | Units: 'm'
Config Value - Param: 'soil_params.b' | Value: '8.93396282196045' | Units: ''
Config Value - Param: 'soil_params.satdk' | Value: '3.19069084890877e-05' | Units: 'm s-1'
Config Value - Param: 'soil_params.satpsi' | Value: '3.98730560956446' | Units: 'm'
Config Value - Param: 'soil_params.slop' | Value: '0.057029859113015' | Units: 'm/m'
Config Value - Param: 'soil_params.smcmax' | Value: '0.401686143900526' | Units: 'm/m'
Config Value - Param: 'soil_params.wltsmc' | Value: '0.048334490431746' | Units: 'm/m'
Config Value - Param: 'soil_params.expon' | Value: '1.0' | Units: ''
Config Value - Param: 'soil_params.expon_secondary' | Value: '1.0' | Units: ''
Config Value - Param: 'refkdt' | Value: '3.72730851635058' | Units: '(null)'
Config Value - Param: 'max_gw_storage' | Value: '0.016' | Units: 'm'
Config Value - Param: 'Cgw' | Value: '0.0018' | Units: 'm h-1'
Config Value - Param: 'expon' | Value: '6.0' | Units: ''
Config Value - Param: 'gw_storage' | Value: '0.05' | Units: 'm/m'
Config Value - Param: 'alpha_fc' | Value: '0.33' | Units: '(null)'
Config Value - Param: 'soil_storage' | Value: '0.05' | Units: 'm/m'
Config Value - Param: 'K_nash' | Value: '0.03' | Units: ''
Config Value - Param: 'K_lf' | Value: '0.01' | Units: ''
Config Value - Param: 'nash_storage' | Value: '0.0,0.0' | Units: '(null)'
Config Value - Param: 'num_timesteps' | Value: '1' | Units: '(null)'
Config Value - Param: 'verbosity' | Value: '1' | Units: '(null)'
Config Value - Param: 'DEBUG' | Value: '0' | Units: '(null)'
Config Value - Param: 'giuh_ordinates' | Value: '1.00,0.00' | Units: '(null)'
Found configured GIUH ordinate values ('1.00,0.00')
Config Value - Param: '' | Value: '(null)' | Units: '(null)'
Config Value - Param: '' | Value: '(null)' | Units: '(null)'
Schaake Magic Constant calculated
All CFE config params present
GIUH ordinates string value found in config ('1.00,0.00')
Counted number of GIUH ordinates (2)
Finished function parsing CFE config
At declaration of smc_profile size, soil_reservoir.n_soil_layers = 0

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 20 RUNNING AT d9fcff177365
=   EXIT CODE: 6
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```

**Comment by shahab122** [comment]
@arpita0911patel can we have data from the datastream scripts for a longer period (e.g., a month to a few months)? It would help us to evaluate the simulated streamflow against the USGS measurements.

**Comment by arpita0911patel** [comment]
@shahab122 - created new ticket to address that here: https://github.com/CIROH-UA/NGIAB-CloudInfra/issues/141

Since the above issue is resolve, we could close this ticket.
